src/nvim/charset.c

```diff
@@ -1598,10 +1609,14 @@ int vim_islower(int c)
   return islower(c);
 }
 
-int vim_isupper(int c)
+/// Check that the character is upper-case
+///
+/// @param  c  character to check
+bool vim_isupper(int c)
+  FUNC_ATTR_CONST FUNC_ATTR_WARN_UNUSED_RESULT FUNC_ATTR_NONNULL_ALL
```

Comment:
Maybe `CONST` is possible after simplifications to our encoding handling, but for now this depends on global state (`p_enc`) and is thus `PURE`